nav2_msgs/action/ComputePathToPose.action

```diff
@@ -7,5 +7,6 @@ bool use_start # If false, use current robot pose as path start, if true, use st
 #result definition
 nav_msgs/Path path
 builtin_interfaces/Duration planning_time
+GlobalPlannerResultCode result_code
```

Comment:
You can define the codes in this message and have the type here just be the int16 (and 16 is probably fine for this).

Start the enumeration at 2 though, leave `0` for invalid and `1` for unknown. 

Also, how did you come up with that list of result codes? Did you go through all the planners and find all the reasons why they throw exceptions / fail? Any others we should add? Might be worth brain storming some ideas for other ones that would be of use